base/sparse/sparsevector.jl

```diff
@@ -25,15 +25,33 @@ end
 SparseVector(n::Integer, nzind::Vector{Ti}, nzval::Vector{Tv}) where {Tv,Ti} =
     SparseVector{Tv,Ti}(n, nzind, nzval)
 
+const AbstractSparseVectorUnion{T<:Number} =
+    Union{AbstractSparseVector{T},
+          SubArray{T,1,<:AbstractSparseArray,Tuple{Base.Slice{Base.OneTo{Int64}},Int64},false}}
```

Comment:
Perhaps `<:AbstractSparseMatrix` rather than `<:AbstractSparseArray`? The `Tuple` defining the slice appears specific to two dimensional arrays. (Perhaps even just `<:SparseMatrixCSC`? The `AbstractSparse[Array|Matrix|Vector]` interfaces are not well defined, and the methods below appear to be written against the `SparseMatrixCSC` interface.)